modules/indexer/code/indexer.go

```diff
@@ -163,9 +107,9 @@ func Init() {
 	// Create the Queue
 	switch setting.Indexer.RepoType {
 	case "bleve", "elasticsearch":
-		handler := func(items ...*IndexerData) (unhandled []*IndexerData) {
-			idx, err := indexer.get()
-			if idx == nil || err != nil {
+		handler := func(items ...*internal.IndexerData) (unhandled []*internal.IndexerData) {
+			indexer := globalIndexer
```

Comment:
Here is a data race. The `globalIndexer` should be `atomic.Value` or `atomic.Pointer`

Reply:
You approved while there is a data race in the new code?

Reply:
Yup, but it's only a very very rare case, and it can be easily fixed (I am sure  wolfogre  can fix it)

Reply:
Fixed it in 11fd0f9895 . Got some trouble with `atomic.Value` because of inconsistently typed value. And `atomic.Pointer` works well.